docs/samples/client/python/oauth/unbranded_auth_code_credential.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, Optional
+from corehttp.credentials import AccessTokenInfo, TokenRequestOptions
+import time
+import httpx
+
+
+class UnbrandedAuthCodeCredential:
+    """A credential that uses auth code to authenticate.
+
+    This credential acquires tokens via auth code flow.
+
+    :param str client_id: The client ID of the application.
+    :param str authorization_code: The authorization code.
+    :keyword str authority_url: The authority URL for token requests.
+    :keyword str redirect_uri: The redirect URI.
+    :keyword str client_secret: The client secret of the application.
+    """
+
+    def __init__(
+        self,
+        client_id: str,
+        authorization_code: str,
+        *,
+        authority_url: Optional[str] = None,
+        redirect_uri: Optional[str] = None,
+        client_secret: Optional[str] = None,
+        **kwargs: Any
+    ):
+        """Create a UnbrandedAuthCodeCredential with the provided client ID and authorization_code.
+
+        :param str client_id: The client ID of the application.
+        :param str authorization_code: The authorization code.
+        :keyword str authority_url: The authority URL for token requests.
+        :keyword str redirect_uri: The redirect URI.
+        :keyword str client_secret: The client secret of the application.
+        """
+        self.client_id = client_id
+        self.authorization_code = authorization_code
+        self.authority_url = authority_url
+        self.redirect_uri = redirect_uri
+        self.client_secret = client_secret
+
+    def get_token_info(self, *scopes: str, options: Optional[TokenRequestOptions] = None) -> AccessTokenInfo:
+        """Get an access token for the specified scopes.
+
+        :param str scopes: The scopes for which the token should be valid.
+        :keyword options: A dictionary of options for the token request. Unknown options will be ignored. Optional.
+        :paramtype options: TokenRequestOptions
+        :return: An access token for the specified scopes.
+        :rtype: ~corehttp.credentials.AccessTokenInfo
+        :raises ValueError: If no scopes are provided or if authority URL is missing
+        """
+        if not scopes:
+            raise ValueError("At least one scope must be provided.")
+
+        authority_url = self.authority_url
+        if options is not None:
```

Comment:
```suggestion
        # Options is None, if there was no auth information in the original TypeSpec
        if options is not None:
```

Don't hesitate to over-comment a sample